app/models/task.py

```diff
@@ -1,5 +1,37 @@
-from sqlalchemy.orm import Mapped, mapped_column
+import datetime
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey, DateTime
+from typing import Optional
 from ..db import db
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[datetime.datetime] = mapped_column(DateTime(), nullable=True, default=None)
```

Comment:
Prefer `completed_at` to also use `Optional[]` without passing `nullable=True` to `mapped_column`, like you do with `goal_id` and `goal`.

```suggestion
    completed_at: Mapped[Optional[datetime.datetime]] = mapped_column(default=None)
```